be/src/agent/task_worker_pool.cpp

```diff
@@ -1024,6 +1025,18 @@ void* TaskWorkerPool::_update_tablet_meta_worker_thread_callback(void* arg_this)
                 case TTabletMetaType::INMEMORY:
                     // This property is no longer supported.
                     break;
+                case TTabletMetaType::ENABLEPERSISTENTINDEX:
+                    LOG(INFO) << "update tablet:" << tablet->tablet_id()
+                              << " enable_persistent_index:" << tablet_meta_info.enable_persistent_index;
+                    tablet->set_enable_persistent_index(tablet_meta_info.enable_persistent_index);
+                    // TODO
+                    // If tablet is doing apply rowset right now, remove primary index from index cache may be failed
+                    // So even though `enable_persistent_index` change successfully, it may not take effect immediately
```

Comment:
You must make the operation succeed smoothly.

Reply:
> You must make the operation succeed smoothly.

done, change in https://github.com/StarRocks/starrocks/pull/4958/commits/a9a2c1bb7ec43eb9e1e076d080e2fdc65b5734d8